### ISSUE TITLE
Inconsistent and buggy rendering using draw_set_transform()

### ISSUE THREAD

**Issue by djrain** [godotengine/godot#49156]
**Godot version:**
3.3.2 stable

**OS/device including version:**
2020 Mac Mini
macOS Big Sur

**Issue description:**
Basically, calling draw_set_transform() outside a loop causes buggy rendering compared to inside the loop, even when it shouldn't make any difference.

In this example scene I have a Polygon2D, a Light2D, and a Node2D (called CircleRenderer). CircleRenderer has the following script which should simply draw a green circle over every point on the Polygon2D. Note that the polygon is positioned at (500, 500), while CircleRenderer is at the origin.

```
extends Node2D
tool

func _process(delta):
	update()

func _draw():
	var poly = $"../Polygon2D"
	#draw_set_transform(poly.global_position, 0, Vector2.ONE)
	for point in poly.polygon:
		draw_set_transform(poly.global_position, 0, Vector2.ONE)
		draw_circle(point, 50, Color.seagreen)
```
 As it is, this code works as expected. 

Now suppose I swap the call to draw_set_transform() with the commented out one. This should have the same effect, except for maybe being more efficient. When I do this however, the circles still draw in the correct positions, but weird things happen with all the circles after the first: the Light2D ceases to interact correctly with them, and also, the circles will suddenly disappear if the editor is zoomed in close enough and/or moved far enough. Both behaviors are shown here:

https://user-images.githubusercontent.com/33777501/119914591-684c9580-bf15-11eb-8839-2eb4a3e7cc4d.mov

Also, for some reason, this only seems to happen if the polygon's position is positive.
 
The rendering at runtime is the same. Issue is present in both GLES2 and 3, with or without batching.

**Steps to reproduce:**
Play with MRP example scene

**Minimal reproduction project:**
[DrawSetTransformBug.zip](https://github.com/godotengine/godot/files/6557607/DrawSetTransformBug.zip)

**Comment by lawnjelly** [comment]
Simpler reproduction project, the poly and the texture rect aren't needed. Just move the 2d light and you get the light going off prematurely as you move it down:
[djrain_DrawSetTransformBug.zip](https://github.com/godotengine/godot/files/6558791/djrain_DrawSetTransformBug.zip)

```
func _draw():
	draw_set_transform(Vector2(0, 500), 0, Vector2.ONE)
	draw_circle(Vector2(500, 0), 50, Color.seagreen)
	draw_circle(Vector2(500, 200), 50, Color.seagreen)
```
I suspect this is something to do with AABBs being calculated wrongly in this situation.

**Comment by lawnjelly** [comment]
The problem is in this test:
```
p_ci->global_rect_cache.intersects_transformed(light->xform_cache, light->rect_cache)
```
The global_rect_cache appears to be incorrect for the transformed circles.

**Comment by lawnjelly** [comment]
Got it.

rasterizer.h, line 1046, in get_rect():
```
				if (found_xform) {
					r = xf.xform(r);
					found_xform = false;
				}
```
This is a loop where it goes through the commands to find the bounding rect. The problem is that is only applying the transform to the first command, then sets `found_xform` back to false, which is incorrect, because the transform also applies to later commands. The fix will probably be to just remove this line. I'll do some testing and see if this works.

**Comment by akien-mga** [comment]
Fixed by #49160.
